src/ansible_creator/subcommands/add.py

```diff
@@ -0,0 +1,118 @@
+"""Definitions for ansible-creator add action."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from ansible_creator.exceptions import CreatorError
+from ansible_creator.templar import Templar
+from ansible_creator.types import TemplateData
+from ansible_creator.utils import Copier, Walker, ask_yes_no
+
+
+if TYPE_CHECKING:
+    from ansible_creator.config import Config
+    from ansible_creator.output import Output
+
+
+class Add:
+    """Class to handle the add subcommand.
+
+    Attributes:
+        common_resources: List of common resources to copy.
+    """
+
+    common_resources = ("common.devfile",)
+
+    def __init__(
+        self: Add,
+        config: Config,
+    ) -> None:
+        """Initialize the add action.
+
+        Args:
+            config: App configuration object.
+        """
+        self._resource_type: str = config.resource_type
+        self._add_path: Path = Path(config.path)
+        self._force = config.force
+        self._overwrite = config.overwrite
+        self._no_overwrite = config.no_overwrite
+        self._creator_version = config.creator_version
+        self._project = config.project
+        self.output: Output = config.output
+        self.templar = Templar()
+
+    def run(self) -> None:
+        """Start scaffolding the resource file."""
+        self._check_add_path()
+        self.output.debug(msg=f"final collection path set to {self._add_path}")
+
+        self._scaffold()
+
+    def _check_add_path(self) -> None:
+        """Validate the provided add path.
+
+        Raises:
+            CreatorError: If the add path does not exist.
+        """
+        if not self._add_path.exists():
+            msg = f"The path {self._add_path} does not exist. Please provide an existing directory."
+            raise CreatorError(msg)
+
+    def _scaffold(self) -> None:
+        """Scaffold the specified resource file.
+
+        Raises:
+            CreatorError: If there are conflicts and overwriting is not allowed.
+        """
+        self.output.debug(f"Started copying {self._project} resource to destination")
+
+        # Set up template data
+        template_data = TemplateData(
+            resource_type=self._resource_type,
+            creator_version=self._creator_version,
+        )
+
+        # Initialize Walker and Copier for file operations
+
+        walker = Walker(
+            resources=self.common_resources,
```

Comment:
Please move the scaffolding of a devfile into it's own fuction so we can extend add later on, the new function can be conditionally called based on the resource type.

Reply:
Hey @cidrblock ! Tried to incorporate this idea.. Please review the latest changes.